compiler/rustc_span/src/hygiene.rs

```diff
@@ -67,7 +67,7 @@ rustc_index::newtype_index! {
 }
 
 /// A unique ID associated with a macro invocation and expansion.
-#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
Can this cause future issues like the PartialOrd/Ord impls for `DefId` do?

Reply:
Maybe? I didn't know `DefId` have such issues.
I can remove this and rely on `sort_by` instead.

Reply:
The issue with DefId is: since the API allows to sort by DefId, some queries used to sort their result by DefId. Since a later compilation may assign different DefIds to corresponding definitions, we would get in another order.
Making this kind of mistake difficult by restricting the API and using `sort_by` would probably be better.

Reply:
Fixed.